kernel_module/uio/uio_ivshmem.c

```diff
@@ -115,14 +140,26 @@ static int request_msix_vectors(struct ivshmem_info *ivs_info, int nvectors)
 			free_msix_vectors(ivs_info, i - 1);
 			goto error;
 		}
+#else
+		err = request_irq(pci_irq_vector(ivs_info->dev,
+			ivs_info->msix_entries[i].entry), ivshmem_msix_handler, 0,
+			ivs_info->msix_names[i], ivs_info->uio);
 
+		if (err < 0) {
+			free_msix_vectors(ivs_info, i - 1);
+			goto error;
+		}
+#endif
 	}
```

Comment:
The only real change is the first argument for request_irq, right? If you factor that out all the rest can be reused in both cases instead of copying it.

Did the return value of request_irq change? Why the change from if (err) to if (err < 0)?